### PR TITLE
[all_tests][core][dashboard] Fix node logical resources reporting with autoscaler v2

### LINKED ISSUE #38332: [core][dashboard][autoscaler] Getting nodes broken when autoscaler v2 enabled
### What happened + What you expected to happen

```
Traceback (most recent call last):
  File "/home/ray/anaconda3/lib/python3.10/site-packages/ray/dashboard/optional_utils.py", line 213, in _update_cache
    response = task.result()
  File "/home/ray/anaconda3/lib/python3.10/site-packages/ray/dashboard/modules/node/node_head.py", line 278, in get_all_nodes
    all_node_summary, nodes_logical_resources = await asyncio.gather(
  File "/home/ray/anaconda3/lib/python3.10/site-packages/ray/dashboard/modules/node/node_head.py", line 261, in get_nodes_logical_resources
    status_dict = json.loads(status_string)
  File "/home/ray/anaconda3/lib/python3.10/json/__init__.py", line 339, in loads
    raise TypeError(f'the JSON object must be str, bytes or bytearray, '
TypeError: the JSON object must be str, bytes or bytearray, not NoneType


```

### Versions / Dependencies

master

### Reproduction script

NA

### Issue Severity

None

### LINKED ISSUE #38404: Release test many_pgs.aws failed
Release test **many_pgs.aws** failed. See https://buildkite.com/ray-project/release-tests-branch/builds/2045#0189f464-53ea-4b2d-9925-05db2ff89290 for more details.

Managed by OSS Test Policy

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?
With autoscaler v2, we no longer have monitor process writing the status into kv value store, instead, we could get the data from GCS directly through `get_cluster_status`, so we use that instead to populate logical resources info. 

Added unit testing. 

## Related issue number
Closes https://github.com/ray-project/ray/issues/38404
Closes https://github.com/ray-project/ray/issues/38332

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: dashboard/modules/node/node_head.py [hunk 1]
```diff
@@ -246,6 +247,35 @@ async def get_node_module_internal_state(self, req) -> aiohttp.web.Response:
...
+
+                usage_dict = {
+                    r.resource_name: (r.used, r.total)
+                    for r in node.resource_usage.usage
+                }
+                per_node_resources[node.node_id] = "\n".join(
+                    parse_usage(usage_dict, verbose=True)
+                )
+
+            return per_node_resources
```
Comment (chaowanggg) [inline dashboard/modules/node/node_head.py:276]: Do you have a example data for per_node_resources? I would like to confirm the frontend would render as expected

Comment (rickyyx) [reply dashboard/modules/node/node_head.py:276]:
```

{
    'nodeLogicalResources': {
            '6eb8d224b678a6793679cc189e7bc3ef787033005bc3de61ad3d29a2': '0.0/1.0 CPU\n0B/27.53GiB memory\n0B/13.76GiB object_store_memory'
    }
}
```

Comment (rickyyx) [reply dashboard/modules/node/node_head.py:276]: also see the added e2e test.

Comment (chaowanggg) [reply dashboard/modules/node/node_head.py:276]: Thanks!

File: dashboard/tests/test_dashboard.py [hunk 2]
```diff
@@ -628,6 +628,48 @@ def get_cluster_status():
...
+        """ray start --no-monitor --head --num-cpus 1 \
+--system-config={"enable_autoscaler_v2":true}""",
+        """ray start --head --num-cpus 1""",
+    ],
+    indirect=True,
+)
+def test_get_nodes_summary(call_ray_start):
+
+    # The sleep is needed since it seems a previous shutdown could be not yet
+    # done when the next test starts. This prevents a previous cluster to be
```
Comment (chaowanggg) [inline dashboard/tests/test_dashboard.py:647]: Good catch! I believe I met the similar issue before

**Comment by chaowanggg** [comment]
LGTM!

**Review by chaowanggg — APPROVED** [review]

**Comment by rickyyx** [comment]
Release test with v2 working with logcial node rsources:
image

**Pull request merged**